src/game.cpp

```diff
@@ -2012,12 +2013,11 @@ input_context game::get_player_input(std::string &action)
                 for (auto iter = SCT.vSCT.rbegin(); iter != SCT.vSCT.rend(); ++iter) {
                     const direction oCurDir = iter->getDirecton();
 
-                    for (int i = 0; i < (int)iter->getText().length(); ++i) {
-                        const int dex = mon_at(iter->getPosX() + i, iter->getPosY());
+                    tripoint tmp( iter->getPosX(), iter->getPosY(), get_levz() );
+                    for( tmp.x = iter->getPosX(); tmp.x < iter->getPosX() + (int)iter->getText().length(); ++tmp.x ) {
+                        const int dex = mon_at( tmp );
 
                         if (dex != -1 && u.sees(zombie(dex))) {
-                            i = -1;
-
```

Comment:
This looks buggy. The inner loop calls `advanceStepOffset`, which changes `iStepOffset`, which changes the result of `getPosX` and `getPosY`. Because the result of `getPosY` is only used once to initialize `tmp`, the change here won't affect the loop.

`i = -1` is  probably used to restart the loop (the location of the on-screen-text has changed), this behavior is completely gone in the code. (The loop checks whether the on-screen-text would override a monster glyph and (if so) moves the text a bit, but it may again override another monster glyph at the new location.)

In short: it's probably better to create a temporary only for the calling of `mon_at`. Btw. the code ignores NPCs as if they didn't exist (it should perhaps use `critter_at`).